enhancements/api-review/add-unqualified-search-registries.md

```diff
@@ -0,0 +1,193 @@
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+
+## Summary
+
+Today, the cluster wide Image CR (images.config.openshift.io cluster) doesn't have a way of setting the
+registries to be used when pulling an image using short name.
+
+With the move the v2 /etc/containers/registries.conf, there is a new
+option called `unqualified-search-registries`. This is the list of registries that the container tools will check when pulling an image using its short name.
+
+## Motivation
+
+Today, when a user wants to pull images using short names, they have to create a machine-config object that creates a file under `/etc/containers/registries.conf.d` with the changed they want for `unqualified-search-registries`. 
+
+Users can currently configure `AllowedRegistries`, `BlockedRegistries`, and `InsecureRegistries` in the cluster wide Image CR. Adding the `unqualified-search-registries` option to this cluster wide Image CR will make short name configuration for users simple and have all the image related configurations in one CR.
+
+
+### Goals
+
+- Enable the user to be able to set `unqualified-search-registries` in the cluster wide Image CR.
+
+### Non-Goals
+
+## Proposal
+
+The Image API is extended by adding an optional `UnqualifiedSearchRegistries` field with type `[]string` to `RegistrySources`:
+
+```go
+// RegistrySources holds cluster-wide information about how to handle the registries config.
+type RegistrySources struct {
+    // ...
+    
+	// unqualifiedSearchRegistries are registries that will be searched when pulling images using short names.
+	// +optional
+	UnqualifiedSearchRegistries []string `json:"unqualifiedSearchRegistries,omitempty"`
+}
+```
+
+The containerRuntimeConfig controller in the MCO already watches the cluster wide images.config.openshift.io CR for the allowed, blocked, and insecure registries. It will now watch for `unqualified-search-registries` and update `/etc/containers/registries.conf` accordingly.
+
+An example images.config.openshift.io CR will look like:
+```yaml
+apiVersion: config.openshift.io/v1
+kind: Image
+metadata:
+  name: cluster
+spec:
+  RegistrySources:
+    UnqualifiedSearchRegistries:
+    - "reg1.io"
+    - "reg2.io"
+    - "reg3.io"
+```
+
+### User Stories
+
+#### As a user, I would like to use image short names when running my workloads
+
+The user can set the `unqualified-search-registries` with a list of registries to check when pulling an image using short names.
+The user can run `oc edit images.config.openshift.io cluster` and add `UnqualifiedSearchRegistries` under `RegistrySources`. Once this is done, the containerRuntimeConfig controller will roll out the changes to the nodes.
+
+### Implementation Details/Notes/Constraints
```

Comment:
Can we talk about the default value for this setting (docker, redhat registry) and what happens to those when this feature is used?

Reply:
Fixed